design/ErrorHandling.md

```diff
@@ -66,47 +74,43 @@ class MyRenderer {
       this.initFallback();
     }
   }
-  async initWebGPU() {
-    await this.ensureDevice();
-    // ... Upload resources, etc.
-  }
   initFallback() { /* try WebGL, 2D Canvas, or other fallback */ }
-  async ensureDevice() {
+  async initWebGPU() {
     // Stop rendering. (If there was already a device, WebGPU calls made before
     // the app notices the device is lost are okay - they are no-ops.)
     this.device = null;
 
     // Keep current adapter (but make a new one if there isn't a current one.)
-    // If we can't get an adapter, ensureDevice rejects and the app falls back.
-    await ensureAdapter();
-
-    try {
-      await ensureDeviceOnCurrentAdapter();
-      // Got a device.
-      return;
-    } catch (e) {
-      console.error("device request failed", e);
-      // That failed; try a new adapter entirely.
+    await tryEnsureDeviceOnCurrentAdapter();
```

Comment:
nit: this call to tryEnsureDeviceOnCurrentAdapter() can be folded inside the loop.

Reply:
The first one tries to reuse the current adapter while the later ones get a new adapter, though maybe this is wrong. I think this code still needs some revision.